src/RL/nn_structures/variableoutputcpnn.jl

```diff
@@ -17,61 +17,42 @@ end
 # Enable the `|> gpu` syntax from Flux
 Flux.@functor VariableOutputCPNN
 
-wears_mask(s::VariableOutputCPNN) = false
```

Comment:
Let's discuss this, I know that this condition is necessary to be able to handle action space of fixed size which is required for batch-learning. Otherwise, `variable_outputCPNN` was originally designed to address problems with a potentially large total_action space (union of all action space), so that unlike FFCPNN, the pipeline only computes the useful approximate Q-value for a fixed variable.

For the moment we don't solve problems that have this specificity,  so we can decide to keep that as it is, but that 'd make `VariableOutputCPNN` really close from `FFCPNN` (except the cross weight value/variable), That'd be nice to analyse the impact of these weights in the performance. 

what do you think fo this @ziadelassal ?

Reply:
Also, we're working on a new CPNN that has cross weight value/variable, we could analyze the impact of these weights then